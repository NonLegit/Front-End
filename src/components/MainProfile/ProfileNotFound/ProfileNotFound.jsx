import { Typography } from '@mui/material';
import { NotFoundBox, Image, BackHomeButton } from './styles';

/**
 * ProfileNotFound
 *
 * @component ProfileNotFound
 * @property {function} redirect - redirect to home page

 * @returns {React.Component} ProfileNotFound
 */

function ProfileNotFound() {
  const redirect = () => {
    window.location.pathname = '/';
  };
  return (
    <NotFoundBox>
      <Image src="https://www.redditstatic.com/desktop2x/img/snoomoji/snoo_thoughtful.png" />
      <Typography sx={{ fontWeight: 700, marginBottom: 2 }}>Sorry, nobody on Reddit goes by that name.</Typography>
      <Typography variant="body2" sx={{ marginBottom: 5 }}>The person may have been banned or the username is incorrect.</Typography>
      <BackHomeButton variant="contained" onClick={redirect}>Go Home</BackHomeButton>
    </NotFoundBox>
  );
}

export default ProfileNotFound;
