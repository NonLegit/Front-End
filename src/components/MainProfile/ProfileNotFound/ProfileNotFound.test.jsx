import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import ProfileNotFound from './ProfileNotFound';

test('test snapshot', async () => {
  const tree = renderer.create(
    <ProfileNotFound />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
