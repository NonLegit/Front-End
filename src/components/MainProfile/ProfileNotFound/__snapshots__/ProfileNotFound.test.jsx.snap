// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot 1`] = `
<div
  className="MuiBox-root css-1v93rv6"
>
  <img
    className="css-1n7dhzd"
    src="https://www.redditstatic.com/desktop2x/img/snoomoji/snoo_thoughtful.png"
  />
  <p
    className="MuiTypography-root MuiTypography-body1 css-9vwyyl-MuiTypography-root"
  >
    Sorry, nobody on Reddit goes by that name.
  </p>
  <p
    className="MuiTypography-root MuiTypography-body2 css-5hdb8b-MuiTypography-root"
  >
    The person may have been banned or the username is incorrect.
  </p>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1fkb2of-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    Go Home
  </button>
</div>
`;
